doc/source/tools/gmsh/gmsh.rst

```diff
@@ -0,0 +1,371 @@
+
+.. code-block::
+
+	// LcMax -                         /------------------
+	//                               /
+	//                             /
+	//                           /
+	// LcMin -o----------------/
+	//        |                |       |
+	//     Attractor       DistMin   DistMax
+
+1. Set the attractor:
+
+.. code-block::
+
+	Field[1] = Attractor;
+	
+2. Specify where the refinement needs to be done (near the circle in this case):
+
+.. code-block::
+
+	Field[1].EdgesList = {5};
+	
+3. Set the minimum/maximum characteristic length and the minimum/maximum distance of the refinement:
+
+.. code-block::
+
+	Field[2] = Threshold;
+	Field[2].IField = 1;
+	Field[2].LcMin = 0.25;
+	Field[2].LcMax = 1;
+	Field[2].DistMin = 1;
+	Field[2].DistMax = 2;
+	
+4. Apply the attractor:
+
+.. code-block::
+
+	Background Field = 2;
+	
+Here is the mesh generated with an attractor around the sphere:
+
+.. image:: images/attractor.png
+    :alt: The mesh generated with an attractor arround the sphere
+    :align: center
+    :name: attractor
+
+.. _structured mesh:
+
+""""""""""""""""""""""""""
+Structured - Quad mesh
+""""""""""""""""""""""""""
+.. warning::
+	The ``.geo`` file must be built with the :ref:`built-in kernel`.
+
+Creating a structured quad-mesh usually take a lot more time than an unstructured quad-mesh, but provides a full control on the mesh generation. To do so:
+
+1. Create the geometry accordingly and add construction elements where needed
+
+.. tip::
+	Converting an unstructured mesh to a structured mesh usually required rewritting a good part of the geometry. Begin by drawing by-hand and check that each of your surfaces have only four points.
+
+2. Define ``Transfinite Line`` (before ``Line Loop``), with:
+	
+.. code-block::
+
+	// Transfinite Loop {<id>, ...} = <number of divisions> Using Progression <num>;
+
+3. Define the ``Line Loop`` (or ``Curve Loop``) and ``Plane Surface`` as for a regular mesh
+
+4. Define ``Transfinite Surface`` (instead of ``Plane Surface``) and recombine them to get a quad mesh:
+
+.. code-block::
+
+	// Recombine Surface {<id>}
+
+5. Generate the mesh
+
+.. seealso::
+  You can find a step-by-step `video <https://youtu.be/1A-b84kloFs?t=316>`_, with a similar geometry (cylinder in flow). 
+
+A mesh can also be partially structured, to better account for a boundary layer for instance: see the ``.geo`` file provided with the :doc:`../../examples/incompressible-flow/2d-transient-around-ahmed-body/2d-transient-around-ahmed-body` example. 
+
+.. _tips:
+
+--------------------------
+Other tips
+--------------------------
+- Use variables and functions to make your ``.geo`` file more adaptative
+	* Example: 
+
+	.. code-block::
+
+		//Parameters
+		L = 5; 			//length
+		C45 = Cos(45*Pi/180); 	//cosine of 45 degrees
+		esf = 1.2; 		//element size factor
+
+		//Points
+		Point(1) = {L, L*C45, 0, esf};
+
+- Use the ``Visibility`` options to get the ID of an element or a physical group easily on the GUI: 
```

Comment:
```suggestion
- Use the ``Visibility`` option to get the ID of an element or a physical group easily on the GUI: 
```

Reply:
It is a tool actually, so I think I am gonna go with tool instead of option.

Reply:
Never mind, it is an option.